src/codec/MapAddEntryListenerToKeyCodec.ts

```diff
@@ -14,103 +14,75 @@
+import {StringCodec} from './builtin/StringCodec';
 import {Data} from '../serialization/Data';
-import {MapMessageType} from './MapMessageType';
-
-var REQUEST_TYPE = MapMessageType.MAP_ADDENTRYLISTENERTOKEY;
-var RESPONSE_TYPE = 104;
-var RETRYABLE = false;
-
-
+import {DataCodec} from './builtin/DataCodec';
+import {UUID} from '../core/UUID';
+import {CodecUtil} from './builtin/CodecUtil';
+
+// hex: 0x011800
+const REQUEST_MESSAGE_TYPE = 71680;
+// hex: 0x011801
+const RESPONSE_MESSAGE_TYPE = 71681;
+// hex: 0x011802
+const EVENT_ENTRY_MESSAGE_TYPE = 71682;
+
+const REQUEST_INCLUDE_VALUE_OFFSET = PARTITION_ID_OFFSET + BitsUtil.INT_SIZE_IN_BYTES;
+const REQUEST_LISTENER_FLAGS_OFFSET = REQUEST_INCLUDE_VALUE_OFFSET + BitsUtil.BOOLEAN_SIZE_IN_BYTES;
+const REQUEST_LOCAL_ONLY_OFFSET = REQUEST_LISTENER_FLAGS_OFFSET + BitsUtil.INT_SIZE_IN_BYTES;
+const REQUEST_INITIAL_FRAME_SIZE = REQUEST_LOCAL_ONLY_OFFSET + BitsUtil.BOOLEAN_SIZE_IN_BYTES;
+const RESPONSE_RESPONSE_OFFSET = RESPONSE_BACKUP_ACKS_OFFSET + BitsUtil.BYTE_SIZE_IN_BYTES;
+const EVENT_ENTRY_EVENT_TYPE_OFFSET = PARTITION_ID_OFFSET + BitsUtil.INT_SIZE_IN_BYTES;
+const EVENT_ENTRY_UUID_OFFSET = EVENT_ENTRY_EVENT_TYPE_OFFSET + BitsUtil.INT_SIZE_IN_BYTES;
+const EVENT_ENTRY_NUMBER_OF_AFFECTED_ENTRIES_OFFSET = EVENT_ENTRY_UUID_OFFSET + BitsUtil.UUID_SIZE_IN_BYTES;
+
+export interface MapAddEntryListenerToKeyResponseParams {
+    response: UUID;
+}
 export class MapAddEntryListenerToKeyCodec {
-
-
-    static calculateSize(name: string, key: Data, includeValue: boolean, listenerFlags: number, localOnly: boolean) {
-// Calculates the request payload size
-        var dataSize: number = 0;
-        dataSize += BitsUtil.calculateSizeString(name);
-        dataSize += BitsUtil.calculateSizeData(key);
-        dataSize += BitsUtil.BOOLEAN_SIZE_IN_BYTES;
-        dataSize += BitsUtil.INT_SIZE_IN_BYTES;
-        dataSize += BitsUtil.BOOLEAN_SIZE_IN_BYTES;
-        return dataSize;
-    }
-
-    static encodeRequest(name: string, key: Data, includeValue: boolean, listenerFlags: number, localOnly: boolean) {
-// Encode request into clientMessage
-        var clientMessage = ClientMessage.newClientMessage(this.calculateSize(name, key, includeValue, listenerFlags, localOnly));
-        clientMessage.setMessageType(REQUEST_TYPE);
-        clientMessage.setRetryable(RETRYABLE);
-        clientMessage.appendString(name);
-        clientMessage.appendData(key);
-        clientMessage.appendBoolean(includeValue);
-        clientMessage.appendInt32(listenerFlags);
-        clientMessage.appendBoolean(localOnly);
-        clientMessage.updateFrameLength();
+    static encodeRequest(name: string, key: Data, includeValue: boolean, listenerFlags: number, localOnly: boolean): ClientMessage {
+        const clientMessage = ClientMessage.createForEncode();
+        clientMessage.setRetryable(false);
+
+        const initialFrame = new Frame(Buffer.allocUnsafe(REQUEST_INITIAL_FRAME_SIZE), UNFRAGMENTED_MESSAGE);
```

Comment:
Is it possible to introduce a static factory method in `Frame` class to avoid `Buffer.allocUnsafe()` call in each codec? My main concern is the `safe-buffer` import that we have now in codecs. As we plan to get rid of `safe-buffer` library it would be nice to avoid having to regenerate all codecs. WDYT?

Reply:
Done. I created the `Frame#createInitialFrame` method with size and the optional flags parameters.